src/library/scala/util/hashing/MurmurHash3.scala

```diff
@@ -101,32 +101,85 @@ private[hashing] class MurmurHash3 {
     h = mixLast(h, c)
     finalizeHash(h, n)
   }
-  /** Compute a hash that depends on the order of its arguments.
-   */
+
+  /** Compute a hash that depends on the order of its arguments. Potential range
+    * hashes are recognized to produce a hash that is compatible with rangeHash.
+    */
   final def orderedHash(xs: IterableOnce[Any], seed: Int): Int = {
     var n = 0
     var h = seed
+    var rangeState = 0 // 0 = no data, 1 = first elem read, 2 = has valid diff, 3 = invalid
+    var rangeDiff = 0
+    var prev = 0
+    var initial = 0
     xs.iterator foreach { x =>
-      h = mix(h, x.##)
+      val hash = x.##
+      h = mix(h, hash)
+      rangeState match {
+        case 0 =>
+          initial = hash
+          rangeState = 1
+        case 1 =>
+          rangeDiff = hash - prev
+          rangeState = 2
+        case 2 =>
+          if(rangeDiff != hash - prev) rangeState = 3
+        case _ =>
+      }
+      prev = hash
       n += 1
     }
-    finalizeHash(h, n)
+    if(rangeState == 2) rangeHash(initial, rangeDiff, prev, seed)
```

Comment:
We could restrict this with `if (seed == seqSeed)`, but I guess either way is fine?

Reply:
There's a `rangeHash` method with a `seed` argument. It's probably best if we always make them compatible, regardless of the seed value.